Add support for `readonly` classes

IonBazan opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      |no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

This change adds support for `readonly` classes generation (PHP 8.2 and up).
It also adds support for `final` classes generation from `ClassReflection` (not sure why it was skipped earlier).

Added PHP 8.2 to the build matrix and `composer.json` - tests are passing.

See: #142

IonBazan committed:
Add support for readonly classes

Signed-off-by: Ion Bazan <ion.bazan@gmail.com>

glo71317 commented:
Hello @Ocramius, I have went through PR and found this PR is supporting PHP8.2, Hence My PR will not be required https://github.com/laminas/laminas-code/pull/151
If possible pls merge this PR on priority after reviewing and release because all tests are green for this and looks fine so that it will be available for community users and i am closing My PR

fezfez commented:
@IonBazan could you rebase on 4.8.x ?

IonBazan commented on src/Generator/ClassGenerator.php:
```diff
@@ -87,6 +88,11 @@ public static function fromReflection(ClassReflection $classReflection)
         }
 
         $cg->setAbstract($classReflection->isAbstract());
+        $cg->setFinal($classReflection->isFinal());
```
@Ocramius this call was missing before - not sure if that was intentional. If not, I think we should backport it to 4.7.x too.

Ocramius replied:
Probably best to patch `4.7.x` with a dedicated test too

IonBazan replied:
Done in #156

Ocramius reviewed (approved):
Rebased, thanks @IonBazan!

Ocramius committed:
Ignoring `::fromArray()` deprecation error in newly added test

This new test limits itself to verifying the semantics
around `final` class creation, and it isn't really a problem
to rely on the deprecated API in here (for now).

Signed-off-by: Marco Pivetta <ocramius@gmail.com>
